driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -326,14 +332,43 @@ private void logServerSelectionFailure(final ServerSelector serverSelector, fina
         }
     }
 
+    @Nullable
+    private ServerTuple selectServer(final ServerSelector serverSelector, final ClusterDescription clusterDescription) {
+        return selectServer(serverSelector, clusterDescription, this::getServer, getRandom());
```

Comment:
I wouldn't mind taking this opportunity to replace the `ThreadLocal<Random>` with `ThreadLocalRandom` now that we've bumped our minimum Java version to 8.

Reply:
Not all drive-by changes are bad, after all ;) Done.

Reply:
:) I'm reminded of one of my favorite quotes, which gives me an out:

"A foolish consistency is the hobgoblin of little minds, adored by little statesmen and philosophers and divines."

Ralph Waldo Emerson